clevertap-core/src/main/java/com/clevertap/android/sdk/pushnotification/PushProviders.java

```diff
@@ -518,70 +478,84 @@ public Void call() {
         //Disable push amp for devices below Api 26
         if (VERSION.SDK_INT < VERSION_CODES.O) {
-            if (existingJobId >= 0) {//cancel already running job
-                jobScheduler.cancel(existingJobId);
-                StorageHelper.putInt(context, Constants.PF_JOB_ID, -1);
-            }
-
-            config.getLogger()
-                    .debug(config.getAccountId(), "Push Amplification feature is not supported below Oreo");
+            config.getLogger().debug(config.getAccountId(), "Pushamp feature is not supported below Oreo");
             return;
         }
 
-        if (jobScheduler == null) {
-            return;
-        }
+        String existingWorkName = StorageHelper.getString(context, Constants.PF_WORK_ID, "");
         int pingFrequency = getPingFrequency(context);
 
-        if (existingJobId < 0 && pingFrequency < 0) {
-            return; //no running job and nothing to create
-        }
-
-        if (pingFrequency < 0) { //running job but hard cancel
-            jobScheduler.cancel(existingJobId);
-            StorageHelper.putInt(context, Constants.PF_JOB_ID, -1);
+        // no running work and nothing to create
+        if (existingWorkName.equals("") && pingFrequency <= 0) {
+            config.getLogger()
+                    .debug(config.getAccountId(), "Pushamp - There is no running work and nothing to create");
             return;
         }
 
-        ComponentName componentName = new ComponentName(context, CTBackgroundJobService.class);
-        boolean needsCreate = (existingJobId < 0 && pingFrequency > 0);
-
-        //running job, no hard cancel so check for diff in ping frequency and recreate if needed
-        JobInfo existingJobInfo = getJobInfo(existingJobId, jobScheduler);
-        if (existingJobInfo != null
-                && existingJobInfo.getIntervalMillis() != pingFrequency * Constants.ONE_MIN_IN_MILLIS) {
-            jobScheduler.cancel(existingJobId);
-            StorageHelper.putInt(context, Constants.PF_JOB_ID, -1);
-            needsCreate = true;
+        // Running work exists but hard cancel
+        if (pingFrequency <= 0) {
+            config.getLogger()
+                    .debug(config.getAccountId(), "Pushamp - Cancelling worker as pingFrequency <=0 ");
+            stopWorker();
+            return;
         }
 
-        if (needsCreate) {
-            int jobid = config.getAccountId().hashCode();
-            JobInfo.Builder builder = new JobInfo.Builder(jobid, componentName);
-            builder.setRequiredNetworkType(JobInfo.NETWORK_TYPE_ANY);
-            builder.setRequiresCharging(false);
-
-            builder.setPeriodic(pingFrequency * Constants.ONE_MIN_IN_MILLIS, 5 * Constants.ONE_MIN_IN_MILLIS);
-            builder.setRequiresBatteryNotLow(true);
-
-            if (Utils.hasPermission(context, "android.permission.RECEIVE_BOOT_COMPLETED")) {
-                builder.setPersisted(true);
```

Comment:
is this related to phone restart and is this handled?

Reply:
WorkManager automatically handles this on it's own
https://stackoverflow.com/questions/50510559/androidx-workmanager-how-to-make-a-scheduled-job-persistent